pgmpy/tests/test_models/test_MarkovChain.py

```diff
@@ -107,6 +107,16 @@ def test_add_variable_new(self):
         self.assertEqual(model.cardinalities['p'], 3)
         self.assertDictEqual(model.transition_models['p'], {})
 
+    def test_copy(self):
+        model = MC(self.variables, self.card, self.start_state)
+        copy = model.copy()
+        self.assertIsInstance(copy, MC)
+        self.assertIsNot(copy, model)
+        self.assertListEqual(hf.recursive_sorted(model.variables), hf.recursive_sorted(copy.variables))
+        self.assertDictEqual(model.cardinalities, copy.cardinalities)
+        self.assertDictEqual(model.transition_models, copy.transition_models)
+        self.assertListEqual(model.state, copy.state)
```

Comment:
@anchitjain1234 You will need to add more tests as @khalibartan has mentioned to check that the values don't change in the copy object if you change something in the original one.